src/DocumentFormat.OpenXml.Features/CalcChain/CalculationChainExtensions.cs

```diff
@@ -0,0 +1,166 @@
+/// </summary>
+public static class CalculationChainExtensions
+{
+    /// <summary>
+    /// Adds calculation chain tracking that can be controlled via <see cref="ICalculationChainFeature"/> accessible via <see cref="SpreadsheetDocument.Features"/>.
+    /// </summary>
+    /// <param name="doc">Document to add calculation chain tracking to.</param>
+    public static void AddCalculationChainChangeTracking(this SpreadsheetDocument doc)
+    {
+        if (doc is null)
+        {
+            throw new ArgumentNullException(nameof(doc));
+        }
+
+        if (doc.Features.Get<ICalculationChainFeature>() is not null)
+        {
+            return;
+        }
+
+        var calcChain = new CalcChainFeature(doc);
+
+        doc.AddElementEventFeature();
+
+        var events = doc.Features.GetRequired<IElementEventFeature>();
+
+        events.Change += calcChain.Change;
+
+        doc.Features.Set<ICalculationChainFeature>(calcChain);
+    }
+
+    private sealed class CalcChainFeature : ICalculationChainFeature
+    {
+        private readonly SpreadsheetDocument _doc;
+
+        public CalcChainFeature(SpreadsheetDocument doc)
+        {
+            _doc = doc;
+        }
+
+        public bool IsPaused { get; set; }
+
+        IEnumerable<CalculationCell> ICalculationChainFeature.Chain
+        {
+            get
+            {
+                if (_doc.WorkbookPart is not { } workbook)
+                {
+                    yield break;
+                }
+
+                if (workbook.CalculationChainPart is not { } part)
+                {
+                    yield break;
+                }
+
+                if (part.CalculationChain is not { } chain)
+                {
+                    yield break;
+                }
+
+                foreach (var child in chain.ChildElements)
+                {
+                    if (child is CalculationCell cell)
+                    {
+                        yield return cell;
+                    }
+                }
+            }
+        }
+
+        internal void Change(FeatureEventArgs<PartElementEventArgs> args)
+        {
+            if (IsPaused)
+            {
+                return;
+            }
+
+            if (args.Type is EventType.Added or EventType.Removing &&
+                args.Argument.Element is CellType { CellReference: { HasValue: true } r, CellFormula: { } } c)
+            {
+                Handle(c, r, args.Type == EventType.Added);
+            }
+        }
+
+        private void Handle(CellType c, StringValue r, bool isAdding)
+        {
+            var worksheet = GetWorksheet(c);
+
+            if (worksheet is null)
+            {
+                return;
+            }
+
+            var chain = GetCalculationChain();
+
+            var existing = Find(chain, r);
+
+            if (!isAdding && existing is not null)
+            {
+                existing.Remove();
```

Comment:
@twsouthwick so after more testing, it looks like it's possible that removal of formulas is the only action that will trigger calcChain.xml part validation and subsequent errors in Excel. This piece would probably deal with the issue in #1457 but I still think we should follow the Excel team guidance to recommend removal of the calcChain.xml part. Adding a formula doesn't appear to trigger the validation in Excel.

Reply:
@twsouthwick could we instead use this trigger to remove the whole calcChain.xml part in response to adding or removing or modifying formulas in cells? I think that would be a better action to take.